packages/legend-application-studio/src/stores/editor/editor-state/element-editor-state/dataProduct/DataProductEditorState.ts

```diff
@@ -393,13 +482,22 @@ export class DataProductEditorState extends ElementEditorState {
     group.id = id;
     group.description = description;
     dataProduct_addAccessPointGroup(this.product, group);
-    return new AccessPointGroupState(group, this);
+    const newGroupState = new AccessPointGroupState(group, this);
+    this.addGroupState(newGroupState);
+    return newGroupState;
   }
 
   deleteAccessPointGroup(val: AccessPointGroupState): void {
     const state = this.accessPointGroupStates.find((a) => a === val);
     runInAction(() => {
-      deleteEntry(this.accessPointGroupStates, state);
+      if (state) {
+        this.deleteGroupState(state);
+        if (state === this.selectedGroupState) {
+          this.setSelectedGroupState(
+            this.accessPointGroupStates.find((a) => a.value.id === 'default'),
```

Comment:
we shouldnt have a reference to 'default` anywhere in the codebase besides creating a new DataProduct with one default group